scrapenews/spiders/rekordnorth.py

```diff
@@ -0,0 +1,143 @@
+            'rekordnorth.co.za/category/opinion-edition/blogs/simply-delicious/',
+            'rekordnorth.co.za/category/lifestyle-news/competitions-in-pretoria-north/',
+            'rekordnorth.co.za/category/lifestyle-news/entertainment-news/',
+            'rekordnorth.co.za/events/event/',
+            'rekordnorth.co.za/category/news-headlines/',
+            'rekordnorth.co.za/category/sports-news/',
+            'rekordnorth.co.za/category/opinion-edition/',
+            'rekordnorth.co.za/category/lifestyle-news/',
+            'rekordnorth.co.za/online-classifieds/',
+            'rekordnorth.co.za/place-ad/',
+            'rekordnorth.co.za/terms-and-conditions/',
+            'rekordnorth.co.za/privacy-policy/',
+            'rekordnorth.co.za/builders/',
+            'rekordnorth.co.za/property/for-sale/',
+            'rekordnorth.co.za/property/to-rent/',
+            'rekordnorth.co.za/rest-assured/',
+            'rekordeast.co.za/parenting/',
+            'rekordnorth.co.za/support-local-business/',
+            'rekordnorth.co.za/i-love-my-city/',
+            'rekordnorth.co.za/?p=128157',
+            'rekordnorth.co.za/student-living/',
+            'rekordnorth.co.za/be-your-best/',
+            'rekordnorth.co.za/magalieskruin-centre/',
+            'rekordnorth.co.za/about-us',
+            '//biz.rekordnorth.co.za/',
+            '//www.guzzle.co.za/',
+            '//www.autodealer.co.za/',
+            '//iab.com/',
+            '//www.bestofpretoria.co.za/',
+            '//localnewsnetwork.co.za/',
+            '//www.caxton.co.za/',
+            '//facebook.com/',
+            '//twitter.com/',
+            '//southcoastherald.co.za/',
+            '//rekordcenturion.co.za/',
+            '//kemptonexpress.co.za/',
+            '//northglennews.co.za/',
+            '//albertonrecord.co.za/',
+            '//ladysmithgazette.co.za/',
+            '//witbanknews.co.za/',
+            '//roodepoortrecord.co.za/',
+            '//southerncourier.co.za/',
+            '//krugersdorpnews.co.za/',
+        )
+    )
+
+    rules = (
+        Rule(link_extractor, process_links='filter_links', callback='parse_item', follow=True),
+    )
+
+    publication_name = 'Rekord Pretoria-North'
+
+    def parse_item(self, response):
+
+        canonical_url = response.xpath('//link[@rel="canonical"]/@href').extract_first()
+        title = response.xpath('//h1[@class="entry-title"]/text()').extract_first()
+        self.logger.info('%s %s', response.url, title)
+        # should we be using canonical_url instead of response.url for the above?
+        og_type = response.xpath('//meta[@property="og:type"]/@content').extract_first()
+        if og_type == 'article':
+            article_body = response.css('div.entry-content')
+            body_html = " ".join(article_body.xpath('//p').extract())
+            byline = response.css('div.author-name').css('::text').extract()
```

Comment:
this should be a string, not an array of strings - you could probably just change it to `extract_first()` - perhaps I should add some validation to give better feedback to contributors.